include/net/ppp.h

```diff
@@ -400,9 +404,12 @@ struct ppp_context {
 
 		/** Magic-Number value */
 		uint32_t magic;
+#if defined(CONFIG_PPP_DIALUP)
+		struct ppp_my_option_data my_options_data[LCP_NUM_MY_OPTIONS];
+#endif
 	} lcp;
 
-#if defined(CONFIG_NET_IPV4)
+#if defined(CONFIG_NET_IPV4) || defined(CONFIG_PPP_DIALUP)
```

Comment:
This looks suspicious, what happens if user disables IPv4 but enables `PPP_DIALUP`, wouldn't the IPCP stop working properly after that?
If you only want to enable PPP_DIALUP if IPv4 is enabled, just add `depends on NET_IPV4` to Kconfig option.

Reply:
Intention here was that you can use PPP_DIALUP without NET_IPV4. That was my original way at the beginning. But yes, maybe this should be changed to be dependent on NET_IPV4.

Reply:
Fixed